prog/weaver/main.go

```diff
@@ -223,19 +229,20 @@ func main() {
 
 	mflag.Parse()
 
+	if justVersion {
+		fmt.Printf("weave %s\n", version)
+		os.Exit(0)
+	}
+
 	peers = mflag.Args()
 	if resume && len(peers) > 0 {
 		Log.Fatalf("You must not specify an initial peer list in conjunction with --resume")
 	}
 
 	common.SetLogLevel(logLevel)
-
-	if justVersion {
-		fmt.Printf("weave router %s\n", version)
-		os.Exit(0)
-	}
-
 	Log.Println("Command line options:", options())
+	Log.Infoln("Command line arguments:", strings.Join(os.Args[1:], " "))
```

Comment:
We can't do this - it could print out the password.  I think the line before should cover the requirement.